### PR TITLE
Ensure workload_entry appender does not process nodes outside of the requested namespace

### PR DESCRIPTION
**Pull request by jshaughn**
Ensure workload_entry appender does not process nodes outside of the requested namespace being processed by the appender.  When querying for istio config you may hit a "leaf" node (for the entire graph), that is inaccessible or excluded from the UI.

This issue was introduced in v1.40 with the new workload_entry appender.

Fixes https://github.com/kiali/kiali/issues/4384

To Recreate/Test:
1. install travel demo
2. select `travel-agency` namespace and verify graph includes incoming traffic from `travel-portal` nodes.  And that those `travel-portal` nodes are accessible (you can select, double-click, right-click etc).
3. update the  CR to exclude the `travel-portal` namespace
```
spec:
  api:
    namespaces:
      exclude:
        - istio-operator
        - kube-.*
        - openshift.*
        - ibm.*
        - kiali-operator
        - travel-portal
```
4. After the CR is reconciled the graph generation for `travel-agency` should cause an error.
5. apply the PR
6. graph generation for `travel-agency` should succeed but the `travel-portal` nodes should now be marked as restricted (key background image).

### CODE REVIEW

File: graph/telemetry/istio/appender/workload_entry.go [hunk 1]
```diff
@@ -39,6 +39,11 @@ func (a WorkloadEntryAppender) applyWorkloadEntries(trafficMap graph.TrafficMap,
 	versionLabel := config.Get().IstioLabels.VersionLabelName
 
 	for _, n := range trafficMap {
+		// Skip the check if this node is outside the requested namespace, we limit badging to the requested namespaces
+		if n.Namespace != namespaceInfo.Namespace {
+			continue
+		}
```
Comment (nrfox) [inline graph/telemetry/istio/appender/workload_entry.go:45]: Is it simple to add a test for this?

Comment (jshaughn) [reply graph/telemetry/istio/appender/workload_entry.go:45]: done.

**Review by nrfox — APPROVED** [review]
LGTM, good catch!

**Comment by pbajjuri20** [comment]
Before PR -> Exclude the travel-portal in Kiali_CR. It through an error in the graph

before_PR_travel


After PR -> Exclude the travel portal in Kiali_CR. The graph is showing perfectly. 

after_pr_travel

**Review by pbajjuri20 — APPROVED** [review]

**Comment by pbajjuri20** [comment]
Tested on Kiali UI 1.41.0 (51b548beaf7ee8e2ddb501f99b1c8f82836e9910) and Kiali Server v1.41.0-SNAPSHOT (3601e4146a6315091c115933b8855c9bab814d0a)

**Comment by sathieu** [comment]
Thanks @jshaughn!

**Comment by jshaughn** [comment]
@sathieu Your welcome, thanks for reporting the problem.  Should be released in 1.41 on Friday.

**Pull request merged**